Create the $GALAXY_MEMORY_MB tool job environment variable

natefoo opened the pull request:
And attempt to set it automatically under Slurm. See my [malloc testing tool](https://github.com/natefoo/galaxy-tools-devtest/blob/master/tools/malloc.xml) for a usage example. Fixes #4741.

natefoo committed:
Create the $GALAXY_MEMORY_MB tool job environment variable and attempt
to set it automatically under Slurm.

jmchilton commented on lib/galaxy/jobs/runners/util/job_script/MEMORY_STATEMENT.sh:
```diff
@@ -0,0 +1,4 @@
+if [ -n "$SLURM_JOB_ID" ]; then
+    GALAXY_MEMORY_MB=`scontrol -do show job "$SLURM_JOB_ID" | sed 's/.*\( \|^\)Mem=\([0-9][0-9]*\)\( \|$\).*/\2/p;d'`
```
I thought you were going to go out of your way to hide problems if this didn't work automatically?

As it stands now - this can cause jobs to fail by spitting out things to stderr right? Or is that assumption wrong?

natefoo replied:
I was working under the false assumption that only the tool's stderr counts anymore. Perhaps that should be the case (in a separate PR)?

natefoo committed:
When setting $GALAXY_MEMORY_MB, redirect stderr to memory_statement.log
in the job directory.

Also, use -1 as the default for the final test to see if
$GALAXY_MEMORY_MB is a positive integer. This allows an admin to
differentiate between a legit case where we don't set $GALAXY_MEMORY_MB
(currently any DRM other than Slurm) and where the command ran but
returned nothing on stdout.
